src/openzaak/components/documenten/models.py

```diff
@@ -228,10 +238,48 @@ def __str__(self):
 
     @property
     def latest_version(self):
+        if settings.CMIS_ENABLED:
+            raise RecursionError(
+                "Using latest_version() with CMIS enabled causes an infinite loop."
+            )
         # there is implicit sorting by versie desc in EnkelvoudigInformatieObject.Meta.ordering
         versies = self.enkelvoudiginformatieobject_set.all()
         return versies.first()
 
+    def save(self, *args, **kwargs):
+        if not settings.CMIS_ENABLED:
+            return super().save(*args, **kwargs)
+        else:
+            pass
+
+    def delete(self, *args, **kwargs):
+        if not settings.CMIS_ENABLED:
+            return super().delete(*args, **kwargs)
+        else:
+            pass
+
+    def lock_document(self, doc_uuid):
+        if settings.CMIS_ENABLED:
+            cmis_storage = CMISDRCStorageBackend()
+            self.lock = cmis_storage.lock_document(uuid=doc_uuid)
+        else:
+            self.lock = _uuid.uuid4().hex
+
+    def unlock_document(self, doc_uuid, lock, force_unlock=False):
+        if settings.CMIS_ENABLED:
+            cmis_storage = CMISDRCStorageBackend()
+            cmis_storage.unlock_document(uuid=doc_uuid, lock=lock, force=force_unlock)
+        self.lock = ""
+
+    def get_latest_version(self, parent):
```

Comment:
Indeed, `.get_latest_version()` was only called from the `.get_url()` in `EnkelvoudigInformatieObjectHyperlinkedRelatedField`. I decided to remove it and retrieve the EIO directly in the `.get_url()` method because I think it makes a bit more clear what's happening.